src/buffer/out/TextAttribute.hpp

```diff
@@ -165,13 +156,13 @@ class TextAttribute final
 
     // This returns whether this attribute, if printed directly next to another attribute, for the space
     // character, would look identical to the other one.
-    constexpr bool HasIdenticalVisualRepresentationForBlankSpace(const TextAttribute& other, const bool inverted = false) const noexcept
+    bool HasIdenticalVisualRepresentationForBlankSpace(const TextAttribute& other, const bool inverted = false) const noexcept
     {
         // sneaky-sneaky: I'm using xor here
         // inverted is whether there's a global invert; Reverse is a local one.
         // global ^ local == true : the background attribute is actually the visible foreground, so we care about the foregrounds being identical
         // global ^ local == false: the foreground attribute is the visible foreground, so we care about the backgrounds being identical
-        const auto checkForeground = (inverted != _IsReverseVideo());
+        const auto checkForeground = (inverted != IsReverseVideo());

```

Comment:
Wondering if we should move more of the getters into the header and mark them as constexprable instead of stripping it off. Thoughts?

Reply:
I'm glad you brought this up, because I was actually meaning to propose moving _more_ methods out of the header. I don't know if I'm missing something, but I couldn't see any reason why some methods were defined inline and others weren't - it seemed to me to be entirely arbitrary. And other than maybe the constructors, I couldn't see any benefit in making these methods `constexpr` if they're not actually being evaluated as constants anywhere.

That said, I don't feel strongly about it either way, but I would like to understand the reasoning behind the decision. Like what criteria do we use to decide whether a method should be defined inline in the header or not?